### PR TITLE
support extraSecretMounts in query resource

### PR DESCRIPTION
**Pull request by hd-deman**
So we can attach elatic search tls key for example

### CODE REVIEW

**Comment by naseemkullah** [comment]
Thanks for the PR @hd-deman !

We'd also need to add the following under `volumes`:

```
      {{- range .Values.query.extraSecretMounts }}
        - name: {{ .name }}
          secret:
            secretName: {{ .configMap }}
      {{- end }}
```

Also please sign the DCO.

**Comment by hd-deman** [comment]
@naseemkullah fixed

**Comment by naseemkullah** [comment]
Thanks @hd-deman ... Sorry I forgot to mention one last thing, commits [need to be signed](https://help.github.com/en/github/authenticating-to-github/about-commit-signature-verification). Then this is good to go! 🚀

File: charts/jaeger/templates/query-deploy.yaml [hunk 1]
```diff
@@ -189,6 +195,11 @@ spec:
           configMap:
             name: {{ .configMap }}
       {{- end }}
+      {{- range .Values.query.extraConfigmapMounts }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/query-deploy.yaml]:
```suggestion
      {{- range .Values.query.extraSecretMounts }}
```

File: charts/jaeger/templates/query-deploy.yaml [hunk 2]
```diff
@@ -189,6 +195,11 @@ spec:
...
+        - name: {{ .name }}
+          configMap:
```
Comment (naseemkullah) [inline charts/jaeger/templates/query-deploy.yaml]:
```suggestion
          secret:
```

File: charts/jaeger/templates/query-deploy.yaml [hunk 3]
```diff
@@ -189,6 +195,11 @@ spec:
...
+            name: {{ .configMap }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/query-deploy.yaml]:
```suggestion
            secret: {{ .secretName }}
```

**Comment by hd-deman** [comment]
@naseemkullah Can you point me why tests fails?

**Comment by hd-deman** [comment]
I just squash all my commits

**Comment by hd-deman** [comment]
@naseemkullah if `query.agentSidecar` is disabled, it still define volumes and faild with error
```
failed to upgrade chart for release [jaeger]: Deployment.apps "jaeger-query" is invalid: spec.template.spec.volumes[1].name: Duplicate value: "certificates"
```

```
    agent:
      cmdlineParams:
        es.tls.ca: /es/certificates/ca.crt
      extraSecretMounts:
        - name: certificates
          mountPath: /es/certificates/
          secretName: quickstart-es-http-certs-public
          readOnly: true
          subPath: ""

    query:
      agentSidecar:
        enabled: false
      cmdlineParams:
        es.tls.ca: /es/certificates/ca.crt
      extraSecretMounts:
        - name: certificates
          mountPath: /es/certificates/
          secretName: quickstart-es-http-certs-public
          readOnly: true
          subPath: ""
```

**Comment by naseemkullah** [comment]
Error seems to be:

```
 .github:1

reference is not a tree: 6544cd94828700a19478faae3ec09d1adb1ff828
 .github:1

Git checkout failed with exit code: 128
 .github:1

Exit code 1 returned from process: file name '/home/runner/runn...
```


@scottrigby Any idea what might be the issue?

**Comment by naseemkullah** [comment]
Hi @hd-deman, this repo requires that commits be gpg signed in order to merge PRs, could you please gpg sign your commits? 🙏

**Comment by hd-deman** [comment]
@naseemkullah done

**Comment by naseemkullah** [comment]
Hi @hd-deman seems like they are still not gpg signed: https://help.github.com/en/github/authenticating-to-github/about-commit-signature-verification

**Comment by hd-deman** [comment]
No way. check it please 
https://github.com/jaegertracing/helm-charts/pull/56/commits
Screenshot 2020-03-03 at 00 21 31

**Comment by naseemkullah** [comment]
> No way. check it please
> https://github.com/jaegertracing/helm-charts/pull/56/commits
> Screenshot 2020-03-03 at 00 21 31

I think when they are GPG signed you'll have the "Verified" button next to the git commit as follows:

image

P.S. there is a conflict int Chart.yaml since a PR has been merged recently, could you please fix it? 🙏

**Comment by hd-deman** [comment]
@naseemkullah thanks. finally got it

**Comment by naseemkullah** [comment]
Great, thanks @hd-deman ! Hopefully CI passes and we can finally merge this 🤞

**Comment by naseemkullah** [comment]
Seems like CI is experiencing: actions/checkout#23
Could you please try:

git commit --allow-empty -m "rerun build"
?
Might help. 🤷‍♂

**Comment by naseemkullah** [comment]
😭 
@hd-deman Could you try the re-run again? I really hope to improve the CI soon.

**Comment by naseemkullah** [comment]
Bypassing broken CI.

**Pull request merged**